### PR TITLE
Created User API scaffold

### PR DESCRIPTION
**Pull request by chitranshubisht**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -38,6 +38,13 @@ gem "bootsnap", require: false
 group :development, :test do
   # See https://guides.rubyonrails.org/debugging_rails_applications.html#debugging-with-the-debug-gem
   gem "debug", platforms: %i[ mri mswin mswin64 mingw x64_mingw ]
+  gem 'byebug'
+  gem 'rspec-rails'
+  gem 'rspec'
+  gem 'factory_bot'
+  gem 'capybara'
+  gem 'shoulda-matchers'
+  gem 'capybara'
```
Comment (nyadav-deqode) [inline Gemfile]: What is the purpose of Capybara gem?

Comment (nyadav-deqode) [reply Gemfile]: Same for FactoryBot

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+      user = User.new(permitted_params)
+      if user.save
+        render json: user, status: ok
+      else
+        render json: user.errors, status: unprocessable_entity
+      end
+  end
+
+  def index 
+    users = User.all
```
Comment (nyadav-deqode) [inline app/controllers/users_controller.rb]: Update this endpoint to return User posts as well if any

File: config/database.yml [hunk 3]
```diff
@@ -56,6 +56,9 @@ development:
 test:
   <<: *default
   database: user_api_test
+  username: user_api
+  # The password associated with the PostgreSQL role (username).
+  password: user_api
```
Comment (nyadav-deqode) [inline config/database.yml]: Do not push credentials, use ENV instead

File: spec/controllers/posts_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        post :create, params: { post: { title: nil, description: 'Test Description' } }
+        expect(response).to have_http_status(:unprocessable_entity)
+      end
+    end
+  end
+
+  describe 'GET #index' do
+    it 'returns a list of posts' do
+      # Create posts directly in the database
+      Post.create(title: 'Title 1', description: 'Description 1')
```
Comment (nyadav-deqode) [inline spec/controllers/posts_controller_spec.rb:35]: Use Factory to create data

Comment (chitranshubisht) [reply spec/controllers/posts_controller_spec.rb:35]: I have inlcuded this in `feature/authentication-using-jwt-token` branch

File: config/database.yml [hunk 5]
```diff
@@ -21,25 +21,25 @@ default: &default
 
 development:
   <<: *default
-  database: user_api_development
+  database: <%= ENV['USER_DATABASE'] %>
```
Comment (nyadav-deqode) [inline config/database.yml]: No need to use ENV for database name

File: app/controllers/users_controller.rb [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+    if user.save
+      render json: user, status: ok
+    else
+      render json: user.errors, status: unprocessable_entity
+    end
+  end
+
+  def index
+    users_posts = User.includes(:posts)
+    render json: users_posts.to_json(include: :posts)
```
Comment (nyadav-deqode) [inline app/controllers/users_controller.rb]: This is unnecessary. use serializers to achieve this

**Review by nyadav-deqode — CHANGES_REQUESTED** [review]

File: app/controllers/users_controller.rb (hunk 6 above)
Comment (chitranshubisht) [reply app/controllers/users_controller.rb]: Okay will do it using serializer

File: config/database.yml (hunk 5 above)
Comment (chitranshubisht) [reply config/database.yml]: In most of the projects they include database name as well, anyway I will remove it.

File: app/controllers/users_controller.rb [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+    if user.save
+      render json: user, status: ok
+    else
+      render json: user.errors, status: unprocessable_entity
+    end
+  end
+
+  def index
+    @users = User.all
+    users = { users: ActiveModelSerializers::SerializableResource.new(@users.all, each_serializer: UserSerializer) }
```
Comment (nyadav-deqode) [inline app/controllers/users_controller.rb]: This is invalid. Serializers doesn't requires this. Please find an alternative

**Review by nyadav-deqode — APPROVED** [review]

**Pull request merged**